packages/maker.js/src/core/maker.ts

```diff
@@ -782,6 +782,22 @@ namespace MakerJs {
         alternateDirection?: boolean;
     }
 
+    /**
+     * A point reference in a path.
+     */
+    export interface IWalkChainPoints extends IWalkPath{
```

Comment:
let's have this extend `IChainLink` instead. We could rename it to `IChainLinkKeyPoint` or something similar.
Through the `link` we can then access the `walkedPath` and `endPoints`. Rename `point` to `keyPoint` and add `ratio` for thr r value.

Reply:
or don't extend anything, and just add `link` property.